clients/cobol-lsp-vscode-extension/src/__tests__/services/copybook/CopybookDownloadServiceTest.spec.ts

```diff
@@ -838,4 +869,135 @@ describe("Tests copybook download service", () => {
       });
     });
   });
+
+  it("checks dsn settings used when no settings provided in processor group definitions", async () => {
+    const spyConfig = jest.spyOn(
+      ProcessorGroups,
+      "loadProcessorGroupCopybookPathsConfig",
+    );
+    spyConfig.mockResolvedValue([]);
+    SettingsService.getDsnPath = jest.fn().mockReturnValue(["dsn"]);
+
+    const downloader = new CopybookDownloadService(
+      "storage-path",
+      zoweExplorerMock,
+      undefined,
+    );
+    downloader["dsnDownloader"]!.downloadCopybook = jest
+      .fn()
+      .mockReturnValue(true);
+    downloader["ussDownloader"]!.downloadCopybook = jest.fn();
+
+    await downloader.downloadCopybook(
+      { name: "copybook", dialect: "COBOL" },
+      "document-uri",
+    );
+    expect(downloader["dsnDownloader"]!.downloadCopybook).toHaveBeenCalledWith(
+      { name: "copybook", dialect: "COBOL" },
+      "dsn",
+      "profile",
+    );
+  });
+
+  it("checks settings in processor groups used first", async () => {
+    const spyConfig = jest.spyOn(
+      ProcessorGroups,
+      "loadProcessorGroupCopybookPathsConfig",
+    );
+    spyConfig.mockResolvedValue([
+      "/libs",
+      { dataset: "procGroupDataset", profile: "procGroupProfile" },
+      { ussFile: "ussFile", profile: "profile" },
+    ]);
+
+    const downloader = new CopybookDownloadService(
+      "storage-path",
+      zoweExplorerMock,
+      undefined,
+    );
+    downloader["dsnDownloader"]!.downloadCopybook = jest
+      .fn()
+      .mockReturnValue(true);
+
+    await downloader.downloadCopybook(
+      { name: "copybook", dialect: "COBOL" },
+      "document-uri",
+    );
+    expect(downloader["dsnDownloader"]!.downloadCopybook).toHaveBeenCalledWith(
+      { name: "copybook", dialect: "COBOL" },
+      "procGroupDataset",
+      "procGroupProfile",
+    );
+    const settingsMockDsn = (SettingsService.getDsnPath = jest.fn());
+    const settingsMockUss = (SettingsService.getDsnPath = jest.fn());
+    expect(settingsMockDsn).toHaveBeenCalledTimes(0);
+    expect(settingsMockUss).toHaveBeenCalledTimes(0);
+  });
+
+  it("checks settings in processor group definitions used in order ", async () => {
+    const spyConfig = jest.spyOn(
+      ProcessorGroups,
+      "loadProcessorGroupCopybookPathsConfig",
+    );
+    spyConfig.mockResolvedValue([
+      "/libs",
+      { dataset: "procGroupDataset", profile: "procGroupProfile" },
+      { ussFile: "ussFile", profile: "profile" },
+    ]);
+
+    const downloader = new CopybookDownloadService(
+      "storage-path",
+      zoweExplorerMock,
+      undefined,
+    );
+    downloader["dsnDownloader"]!.downloadCopybook = jest
+      .fn()
+      .mockReturnValue(false);
+    downloader["ussDownloader"]!.downloadCopybook = jest
+      .fn()
+      .mockReturnValue(true);
```

Comment:
I don't think this test actually proves the tested behavior.
You need to create a situation where the file is in principle available via multiple sources and then verify that the top most is used first.
Ideally, the input configuration should be specified in the inverse order of how the different "downloaders" are executed in the code.